src/runtime/pkg/magent/containerd.go

```diff
@@ -0,0 +1,94 @@
+package magent
+
+import (
+	"context"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/typeurl"
+	"github.com/sirupsen/logrus"
+
+	vc "github.com/kata-containers/kata-containers/src/runtime/virtcontainers"
+	"github.com/kata-containers/kata-containers/src/runtime/virtcontainers/pkg/oci"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func getContainer(containersClient containers.Store, namespace, cid string) (containers.Container, error) {
+	ctx := context.Background()
+	ctx = namespaces.WithNamespace(ctx, namespace)
+	return containersClient.Get(ctx, cid)
+}
+
+// isSandboxContainer return true if the contaienr is a snadbox container.
+func isSandboxContainer(c *containers.Container) bool {
+	// unmarshal from any to spec.
+	v, err := typeurl.UnmarshalAny(c.Spec)
+	if err != nil {
+		logrus.WithError(err).Errorf("failed to Unmarshal container spec")
+		return false
```

Comment:
Why not make the function return `Error` rather than `bool`? That way the function is still returning a boolean (`Error` or `nil`), and the caller can decide what to do with the error value if there is one.

Reply:
If a `nil` represent `true`, then should I define a new `notSandboxContainerError` to distinguish with other `error` s ?